compliance/lib/data_adapter.rego

```diff
@@ -43,11 +43,17 @@ process_name = name {
 
 process_args_list = args_list {
 	is_process
-	args_list := split(input.resource.command, " ")
+
+	# Gets all the process argument of the current process
+	# Expects format as the following: --<key><delimiter><value> for example: --config=a.json
+	# Supported delimiters are space and `=`
+	# Notice that the first argument is always the process path
+	args_list := regex.find_n("^\\S+|--\\S+=\\S+|--\\S+\\s\\S+", input.resource.command, -1)
 }
 
 process_args = args {
-	args := {arg: value | [arg, value] = common.split_key_value(process_args_list[_])}
+    supported_delimiters = ["="," "]
```

Comment:
It would definitely not work.
But, the code before didn't support it either - so it won't be any regression from before.

(Before my change we split the cmd-arguments by using a space character as the delimiter.)

https://github.com/elastic/csp-security-policies/blob/main/compliance/lib/data_adapter.rego#:~:text=%7D-,process_args_list,-%3D%20args_list%20%7B

As for the argument containing spaces:
1. https://github.com/elastic/beats/blob/main/x-pack/osquerybeat/ext/osquery-extension/pkg/proc/cmdline.go - so it seems that the beats code replacing {0} with spaces, which ensures that after every work there is a space.

2. After a long investigation (Nice catch BTW) - It is not common to give a space in an argument value, but it is possible. I opened a task on it - https://github.com/elastic/security-team/issues/3521